### PR TITLE
Sendgrid updates

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

Added identify call and support for deletion

### CODE REVIEW

**Review by mihir-4116 — APPROVED** [review]

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 1]
```diff
@@ -4,104 +4,72 @@ description: >-
  Step-by-step guide on sending your event data from RudderStack to SendGrid.
 ---
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:5]:
```suggestion
aliases: ["SendGrid destination"]
---
```

Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/sendgrid.mdx:5]: as we have SendGrid as a source too.

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 2]
```diff
@@ -4,104 +4,72 @@ description: >-
...
+## Connection settings
 
-* Next, enter the **Template ID**. 
+To successfully configure SendGrid as a destination, you will need to configure the following settings:
 
-<div class="infoBlock">
- 
-To get your template ID or create a new one, go to your <a href="https://app.sendgrid.com/">SendGrid dashboard</a>. Then, click on <strong>Email API</strong> and go to <strong>Dynamic Templates</strong>.
-</div>
+- **API Key**: Enter your SendGrid API Key. Refer to the [Sendgrid documentation](https://app.sendgrid.com/settings/api_keys) for more information on obtaining the API Key.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:31]:
```suggestion
- **API Key**: Enter your SendGrid API Key. Refer to the [Sendgrid documentation](https://docs.sendgrid.com/ui/account-and-settings/api-keys) for more information on obtaining the API Key.
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 3]
```diff
@@ -4,104 +4,72 @@ description: >-
...
 </div>
 
 <div class="infoBlock">
- 
-For more information on how to send these details via the <code class="inline-code">replyTo</code> object, refer to the refer to the <Link to="#replyto">replyTo</Link> section below.
+For more information on how to send these details via the <code class="inline-code">replyTo</code> object, refer to the <Link to="#replyto">replyTo</Link> section below.
 </div>
 
-* Next, enter the **IP Pool Name**.
+- **IP Pool Name**: Enter the IP Pool Name. For detailed steps on creating the IP pool name, refer to the <a href="https://docs.sendgrid.com/ui/account-and-settings/ip-pools#create-an-ip-pool">SendGrid documentation</a>.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:56]:
```suggestion
- **IP Pool Name**: Enter the IP Pool Name. For detailed steps on creating the IP pool, refer to the <a href="https://docs.sendgrid.com/ui/account-and-settings/ip-pools#create-an-ip-pool">SendGrid documentation</a>.
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 4]
```diff
@@ -4,104 +4,72 @@ description: >-
...
 
 <div class="infoBlock">
- 
-The length of the IP Pool Name must be between 2 and 64 characters. For detailed steps on creating the IP pool name, refer to the <a href="https://docs.sendgrid.com/ui/account-and-settings/ip-pools#create-an-ip-pool">SendGrid documentation</a>.
+The length of the IP Pool Name must be between 2 and 64 characters.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:59]:
```suggestion
The length of the IP Pool name must be between 2 and 64 characters. 
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 5]
```diff
@@ -114,22 +82,18 @@ from:{
 
 #### Event names
 
-* Next, create the list of **events** for which the `track` calls will be made.
+- **Event**: Create the list of events for which the `track` calls will be made.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:85]:
```suggestion
- **Event**: Create the list of events for which RudderStack makes the `track` calls.
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 6]
```diff
@@ -114,22 +82,18 @@ from:{
...
 
 <div class="warningBlock">
-
 Any <code class="inline-code">track</code> call made with an event name that is not specified in this list will be <strong>discarded</strong>.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:88]:
```suggestion
RudderStack <strong>discards</strong> any <code class="inline-code">track</code> call made with an event name that is not specified in this list.
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 7]
```diff
@@ -146,18 +110,13 @@ content: [
...
-
-* **type**: The attachment **type** contains the type of content you are attaching e.g. `"text/plain"`, `"text/html"`, etc.
-
-* **filename**: Use this setting to specify the attachment's file name.
-
-* **disposition**: This option specifies how you would like the attachment to be displayed.
-
-* **content ID**: Use this option when **disposition** is set to `inline` and the attachment is an image.
+- **Content**: This should be a Base64-encoded string.
+- **Type**: The attachment **type** contains the type of content you are attaching e.g. `"text/plain"`, `"text/html"`, etc.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:114]:
```suggestion
- **Type**: The attachment **type** contains the type of content you are attaching, for example, `"text/plain"`, `"text/html"`, etc.
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 8]
```diff
@@ -243,6 +200,79 @@ The <strong>utm source</strong>, <strong>utm medium</strong>, <strong>utm term</
 
 Finally, click on **Next** to complete the setup. SendGrid will now be enabled as a destination in RudderStack.
 
+## Identify
+
+You can use the <Link to="/event-spec/standard-events/identify">`identify`</Link> call to create a new contact in SendGrid based on their email ID. If the email ID already exists, RudderStack updates the contact details. 
+
+Also, RudderStack sends the `identify` events to SendGrid in a batch where the batch limit is 30000.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:207]:
```suggestion
RudderStack sends the `identify` events to SendGrid in a batch where the batch limit is 30000.
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 9]
```diff
@@ -243,6 +200,79 @@ The <strong>utm source</strong>, <strong>utm medium</strong>, <strong>utm term</
...
+  "externalId": [{
+    "type": "listIds",
+    "id": ["037ae8d4-25b4-496e-adff-2fded15fd0c5", "137ae8d4-25b4-496e-adff-2fded15fd0c3"]
+  }],
+});
+```
+
+You can pass the SendGrid `listId` in any of the following ways:
+
+- In `externalId` object as shown in the above code snippet (takes highest precendence)
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:225]:
```suggestion
- In `externalId` object as shown in the above code snippet (takes the highest precedence)
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 10]
```diff
@@ -243,6 +200,79 @@ The <strong>utm source</strong>, <strong>utm medium</strong>, <strong>utm term</
...
+- In **SendGrid Contact List Id** Rudderstack dashboard setting.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:226]:
```suggestion
- In **SendGrid Contact List Id** RudderStack dashboard setting.
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 11]
```diff
@@ -243,6 +200,79 @@ The <strong>utm source</strong>, <strong>utm medium</strong>, <strong>utm term</
...
+| `last_name` | `last_name` |
+| `postal_code`   | `postal_code` |
+| `state_province_region` | `state` |
+| `address_line_1`   | `address` |
+| `phone_number` | `phone_number` |
+
+<div class="infoBlock">
+If you send any custom field in the event payload, make sure it is present in SendGrid and mapped in the <strong>Map your traits to SendGrid custom fields</strong> RudderStack dashboard setting.
+</div>
+
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:247]:
```suggestion
```

File: docs/destinations/streaming-destinations/sendgrid.mdx [hunk 12]
```diff
@@ -243,6 +200,79 @@ The <strong>utm source</strong>, <strong>utm medium</strong>, <strong>utm term</
...
+  "users": [{
+      "userId": "eab57ccf-6322-498e-9338-7761c6dc0656"
+    },
+    {
+      "userId": "47983ca9-7cc6-4942-8ff0-ac443g95658d"
+    }
+  ],
+}
+```
+
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/sendgrid.mdx:274]:
```suggestion
```

**Review by ameypv-rudder — APPROVED** [review]

**Pull request merged**